### PR TITLE
Remove unnecessary use of async/await and new Promise

### PR DESCRIPTION
**Pull request by vdvleon**

### CODE REVIEW

File: src/resources/Account.js [hunk 1]
```diff
@@ -25,144 +25,110 @@ export default class Account extends Resource {
     })
   }
 
-  login (email, password = '') {
-    return new Promise(async (resolve, reject) => {
-      try {
-        const salt = await this.retrieveSalt(email) || ''
-        const hash = await bcrypt.hash(password, salt)
-        const result = await this.validateCredentials(email, hash)
+  async login (email, password = '') {
```
Comment (izolate) [inline src/resources/Account.js]: `try/catch` is better here. I plan to normalize all errors eventually, but this pattern should be observed.

For now you can:

```js
try {
  const user = await getUser()
} catch (err) {
  throw err
}

File: src/resources/Account.js [hunk 2]
```diff
@@ -25,144 +25,110 @@ export default class Account extends Resource {
...
-        resolve({
-          id,
-          email,
-          fullName,
-          firstName,
-          lastName
-        })
-      } catch (e) {
-        reject(e.message)
+    const salt = await this.retrieveSalt() || ''
```
Comment (izolate) [inline src/resources/Account.js]: Again, `try/catch`

File: src/resources/Account.js [hunk 3]
```diff
@@ -25,144 +25,110 @@ export default class Account extends Resource {
...
+    return {
+      id,
+      email,
+      fullName,
+      firstName,
+      lastName
+    }
   }
 
   logout () {
```
Comment (izolate) [inline src/resources/Account.js]: This would be better as an async function, without `.then`

File: src/resources/Account.js [hunk 4]
```diff
@@ -25,144 +25,110 @@ export default class Account extends Resource {
...
-          organisations,
-          timezone,
-          language,
-          balance
-        })
-      } catch (e) {
-        reject(e.message)
-      }
-    })
+    return this.request({
```
Comment (izolate) [inline src/resources/Account.js]: Same, async

File: test/workplaces.js [hunk 5]
```diff
@@ -16,11 +16,18 @@ test('Urgency', async t => {
   const start = monthAgo.toISOString()
   const end = now.toISOString()
 
-  const urgency = await deskbookers.workplaces.urgency(17657, [
+  const what = [
```
Comment (izolate) [inline test/workplaces.js]: `what` is non-descriptive and unsightly, please choose another variable name, such as `params`

File: src/resources/Account.js (hunk 1 above)
Comment (vdvleon) [inline src/resources/Account.js]: The thing is, the outcome of the following two pieces of code is identical (ofc), I tested this with the following:

```js
test()
  .then(console.log.bind(console))
  .catch(console.error.bind(console))
```

**Version 1:**
```js
const test = async () => {
  try {
    throw new Error('Test error')
  } catch (e) {
    throw e
  }
}
```

**Version 2:**
```js
const test = async () => {
  throw new Error('Test error')
}
```

If the only reason to do this is to standardize error handling I suggest the following:

- Make a error handler helper, something like:
```js
export default async (callback) => {
  try {
    return await callback()
  } catch (error) {
    throw error
  }
}
```
- When we want to add error tracking we only have to add it in one place.

Now in the SDK functions you do the following:

```js
import errorHandler from '../helpers/error-handler'

// ...
login (email, password) {
  return errorHandler(async () => {
    // ...
  })
}

// ...
```

(Or maybe we can even automatically apply this error handler wrapper to all functions of a resource class)

**Comment by izolate** [comment]
Looks good 🌮

**Pull request merged**